AlgebraInLean/Chapter2/Sheet2.lean

```diff
@@ -0,0 +1,133 @@
+import Mathlib.Tactic
+
+namespace Defs
+
+/-
+
+## Isomorphisms
+
+In Mathlib, isomorphisms come with additional structure; they are not simply defined as bijective
+homomorphisms.
+
+They are defined as a structure, bundled up with some useful fields:
+- to_fun (a map from a group G → group H)
+- inv_fun (a map from group H → group G)
+- left_inv & right_inv (both inverses exist, thus a unique inverse exists)
+- map_mul' (a proof of homomorphism/preservation of operation)
+
+So to prove an isomorphism, we have to provide proofs for each of these fields.
+
+Mathlib also lets us say that two groups are isomorphic by using the symbol `≃+` for additive
+groups, and `≃*` for multiplicative groups.
+
+Using this structure, we can come up with an arbitrary bijection and prove that it is an isomorphism
+(or not), like in the trivial example below. The identity map that takes each element of the
+additive group of integers to itself is clearly an isomorphism. In fact, as you might've guessed,
+this holds for any arbitrary group.
+
+-/
+
+-- Example, not exercise
+-- The identity map is an isomorphism
+example (φ : ℤ → ℤ) (h1 : ∀ x, φ x = x) : ℤ ≃+ ℤ := by
+  let hom_map : ℤ ≃ ℤ := by
+  { constructor
```

Comment:
1. The braces look weird.
2. Use the `change` tactic instead of `have`...`exact`. It looks more elegant and is an opportunity to reinforce concepts introduced earlier in the chapter.